sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/View.java

```diff
@@ -157,7 +166,10 @@ private View() {}
    * PCollectionView} mapping each window to a {@link List} containing all of the elements in the
    * window.
    *
-   * <p>Unlike with {@link #asIterable}, the resulting list is required to fit in memory.
+   * <p>This view should only be used if random access and/or size of the PCollection is required.
+   * {@link #asIterable()} will perform significantly better for sequential access.
```

Comment:
No, most runners will be able to easily support an efficient iterable representation even if they only support an efficient multimap representation since the mapping on the runners side is trivial. The converse is not true.

Also, the list side input adds additional overhead in the format which the iterable does not.

Reply:
Thanks for the explanation.